From ff05570ba755df04961de05e2fc600052ee31216 Mon Sep 17 00:00:00 2001
From: ruslanbay <67730802+ruslanbay@users.noreply.github.com>
Date: Wed, 10 Dec 2025 15:08:31 +0100
Subject: [PATCH 17/54] media: intel/ipu4: Use multiplanar buffer type for format enumeration

Fix format enumeration to use V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE instead of
V4L2_BUF_TYPE_VIDEO_CAPTURE. Since the device is advertised as supporting
multiplanar capture, format enumeration must match this buffer type to be
consistent with userspace expectations.
---
 drivers/media/pci/intel/ipu-isys-video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index d07cb2a6e066..13a10e6005e5 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -441,7 +441,7 @@ int ipu_isys_vidioc_enum_fmt(struct file *file, void *fh,
 		if ((ipu_isys_pfmts[i].is_meta &&
 		     f->type != V4L2_BUF_TYPE_META_CAPTURE) ||
 		    (!ipu_isys_pfmts[i].is_meta &&
-		     f->type != V4L2_BUF_TYPE_VIDEO_CAPTURE))
+		     f->type != V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE))
 			continue;
 
 		if (f->mbus_code && f->mbus_code != ipu_isys_pfmts[i].code)
-- 
2.51.0

